src/create-remote-resource.js

```diff
@@ -1,184 +1,37 @@
-          type: RECEIVE_DATA,
-          now: Date.now(),
-          entryKey: createEntryKey(args),
-          data,
-          resourceId
-        });
-        return data;
-      })
-    );
-  };
-
-  return (...args) => {
-    const entryKey = createEntryKey(args);
-    const { registerError } = useContext(Context);
-    const [entry, setEntry] = useState(selectEntry(entryKey));
-    const data = selectData(entry);
-    const hasSubscription = selectHasSubscription(entry);
-    const cacheInvalid = selectUpdatedAt(entry)
-      .map(updatedAt => updatedAt + invalidateAfter < Date.now())
-      .getOrElse(true);
-
-    useEffect(() => {
+  const resourceId = uuid();
+  return {
+    id: resourceId,
+    createEntryId,
+    initialValue,
+    invalidateAfter,
+    load,
+    save,
+    delete: destroy,
+    getEntry: entryId => selectEntry(store.getState(), { resourceId, entryId }),
+    onChange: onChange => {
+      let currentState = selectResource(store.getState(), { resourceId });
       return store.subscribe(() => {
-        const nextEntry = selectEntry(entryKey);
-        if (nextEntry !== entry) {
-          setEntry(nextEntry);
-        }
-      });
-    }, [entryKey]);
-
-    // We only load on the first render if the cache is invalid
-    const renderCount = useRef(0);
-    renderCount.current = renderCount.current + 1;
-    if (renderCount.current === 1 && cacheInvalid && !loadTasks.has(entryKey)) {
-      load(...args).catch(registerError);
-    }
-
-    // We only suspend while the initial load is outstanding
-    if (cacheInvalid && loadTasks.has(entryKey)) {
-      throw loadTasks.get(entryKey);
-    }
-
-    const setCache = useCallback(
-      valueOrUpdate => {
-        const newData =
-          typeof valueOrUpdate == "function"
-            ? valueOrUpdate(data)
-            : valueOrUpdate;
-        store.dispatch({
-          type: RECEIVE_DATA,
-          now: Date.now(),
-          data: newData,
-          entryKey,
+        const nextResourceState = selectResource(store.getState(), {
           resourceId
         });
-        return newData;
-      },
-      [data]
-    );
-
-    const actions = {
-      refresh: useCallback(() => load(...args).catch(registerError), []),
-      setCache,
-      deleteCache: useCallback(() => {
-        store.dispatch({
-          type: DELETE_ENTRY,
-          entryKey,
-          resourceId
-        });
-        return data;
-      }, [data]),
-      remoteSave: useCallback(
-        newData => saveTasks.run(entryKey, () => save(newData)),
-        []
-      ),
-      remoteDelete: useCallback(
-        () => deleteTasks.run(entryKey, () => destroy(data)),
-        [data]
-      ),
-      subscribe: useCallback(() => {
-        // we only want one subscription running for each entry at a time
-        if (hasSubscription) {
-          return;
+        if (nextResourceState !== currentState) {
+          currentState = nextResourceState;
+          onChange();
```

Comment:
Definitely, but I'm less concerned about that. Since all resources share the same store the function we pass to `store.subscribe` will be called when the state of any of resource changes. This is meant to be a cheap check to avoid an unnecessary re-render if this particular resource was left alone in the store. If they happen to make a lateral change to a resource's state by not changing any of the values i'm fine with letting that go through.